docs/spec/sources.md

```diff
@@ -362,5 +376,39 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
```

Comment:
this is just called `resource`.

Reply:
I added the namespace as part of the value.

@n3wscott I would rather stick with `plural` as that's how it is called in CRDs.

@rhuss this extension attribute is about identifying the event source object that has been used to subscribe to events. For Kamelet, if you want to filter based on the source you connect to, either use the event type or event source. Does this make sense?

Reply:
I would drop all of these and use resource `<uid>-<generation>`